src/pages/ApplicationConfig/components/ComponentDialog/index.tsx

```diff
@@ -0,0 +1,427 @@
+import React from 'react';
+import { Grid, Field, Form, Message, Button, Input, Select, Divider } from '@b-design/ui';
+import type { Rule } from '@alifd/field';
+import { withTranslation } from 'react-i18next';
+import { createApplicationComponent, updateComponentProperties } from '../../../../api/application';
+import { detailComponentDefinition } from '../../../../api/application';
+import type {
+  DefinitionDetail,
+  Trait,
+  ApplicationComponentConfig,
+} from '../../../../interface/application';
+import UISchema from '../../../../components/UISchema';
+import DrawerWithFooter from '../../../../components/Drawer';
+import Translation from '../../../../components/Translation';
+import { checkName } from '../../../../utils/common';
+import Title from '../../../../components/Title';
+import TraitsList from '../../components/TraitsList';
+import { Link } from 'dva/router';
+import locale from '../../../../utils/locale';
+import { If } from 'tsx-control-statements/components';
+import _ from 'lodash';
+import './index.less';
+
+type Props = {
+  appName?: string;
+  componentName?: string;
+  editComponent: any;
+  isEditComponent: boolean;
+  temporaryTraitList: Trait[];
+  componentDefinitions: [];
+  onAddTrait: () => void;
+  changeTraitStats: (is: boolean, params: any) => void;
+  onDeleteTrait: (type: string) => void;
+  onComponentOK: () => void;
+  onComponentClose: () => void;
+  dispatch?: ({}) => {};
+  t: (key: string) => {};
```

Comment:
There is no need to pass this function.  `i18n.t("xxx")`

Reply:
done. use i18.t('xxx')

Reply:
please remove the unuseful code.